app/routes.py

```diff
@@ -0,0 +1,334 @@
+
+@video_bp.route("", methods=["POST"])
+def post_a_video():
+    request_body = request.get_json()
+
+    try:
+        new_video = Video(
+            title = request_body["title"],
+            release_date = request_body["release_date"],
+            total_inventory = request_body["total_inventory"]
+            )
+        db.session.add(new_video)
+        db.session.commit()
+
+        return  {
+            "id" : new_video.id, 
+            "title" : new_video.title, 
+            "release_date" : new_video.release_date,
+            "total_inventory": new_video.total_inventory
+        }, 201
+
+    except KeyError as err:
+        if "title" in err.args:
+            return {"details" : f"Request body must include title."}, 400
+        if "release_date" in err.args:
+            return {"details" : f"Request body must include release_date."}, 400
+        if "total_inventory" in err.args:
+            return {"details" : f"Request body must include total_inventory."}, 400
+
+
+@video_bp.route("/<video_id>", methods=["PUT"])
+def update_a_video(video_id):
+    video = Video.query.get(video_id)
+    request_body = request.get_json()
+    try:
+        video.title = request_body["title"]
+        video.release_date = request_body["release_date"]
+        video.total_inventory = request_body["total_inventory"]
+
+        db.session.add(video)
+        db.session.commit()
+
+        return  {
+            "id" : video.id, 
+            "title" : video.title, 
+            "release_date" : video.release_date,
+            "total_inventory": video.total_inventory
+        }, 200
+    except KeyError:
+        return {'message': "Invalid type"}, 400
+    except:
+        return {"message": f"Video {video_id} was not found"}, 404
+    
+
+@video_bp.route("/<video_id>", methods=["DELETE"])
+def delete_a_video(video_id):
+    try:
+        video = Video.query.get(video_id)
+        if Rental.query.filter_by(video_id=video_id, checked_out=True).count() > 0:
+            return {"message" : f"Video {video_id} has outstanding rentals"}, 200
+        db.session.delete(video)
+        db.session.commit()
+
+        return {
+            "id" : video.id
+        }, 200
+    except:
+        return {"message": f"Video {video_id} was not found"}, 404
+
+
+@rental_bp.route("/check-out", methods=["POST"])
+def check_out_video():
+    request_body = request.get_json()
+    try: 
+        customer_id = request_body["customer_id"]
+        video_id = request_body["video_id"]
+    except KeyError as err:
+        if "customer_id" in err.args:
+            return {"details" : "Missing customer_id"}, 400
+        elif "video_id" in err.args:
+            return {"details" : "Missing video_id"}, 400
+
+    customer = Customer.query.get_or_404(customer_id)
+    video = Video.query.get_or_404(video_id)
+    num_of_rental = Rental.query.filter_by(video_id = video_id).count()
+    available_inventory = video.total_inventory - num_of_rental
```

Comment:
You could add `video_checkout_count` as a column in your customer model then you could do something like 
```
customer.videos_checked_out_count += 1
video.available_inventory -= 1
```